### DISCUSSION TITLE
Best practiace when parsing settings into multiple functions

### DISCUSSION THREAD

**Discussion by xxxpsyduck** [fastapi/fastapi#8877] [Questions]
I read environment variables from a .env file like in your example [here](https://fastapi.tiangolo.com/advanced/settings/?h=en#reading-a-env-file). But I need to parse the `settings` into multiple functions, not just one. Somethings like this:
```python
def A(settings: config.Settings = Depends(get_settings)):
    print(settings.foo)

def B(settings: config.Settings = Depends(get_settings)):
    print(settings.bar)
```
It works only the first time. At the 2nd call in B function, it raises an error: `AttributeError: 'Depends' object has no attribute bar` despite that vscode still suggests `bar` while I'm typing `setting.` I find my practice is kinda bad. Could you suggest a better one?

**Comment by nzig** [comment]
Could you post a more complete code sample? Where is your FastAPI `app`? How are you using these functions?
